tsdb/shard.go

```diff
@@ -43,6 +43,25 @@ var (
 	ErrFieldUnmappedID = errors.New("field ID not mapped")
 )
 
+// A ShardError implements the error interface, and contains extra
+// context about the shard that generated the error.
+type ShardError struct {
+	id  uint64
+	Err error
+}
+
+// NewShardError returns a new ShardError.
+func NewShardError(id uint64, err error) ShardError {
+	if err == nil {
+		err = errors.New("unknown error")
```

Comment:
This changes the meaning of a `nil` error value, and I really don't like that. If you're going to nest errors like this (I don't see the benefit), then at least have the function return the type `error` and if it's passed a `nil` error to be nested, then just return `nil` for this function. I can't imagine how confused I would be if I had to track down an error, and I found out that ultimately, the error value that caused the problem were `nil`.